src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1180,6 +1180,10 @@ memory::ptr primitive_inst::allocate_output(engine& _engine, memory_pool& pool,
     if (_node.is_in_shape_of_subgraph())
         reusable_across_network = false;
 
+    GPU_DEBUG_GET_INSTANCE(debug_config);
+    GPU_DEBUG_IF(debug_config->disable_memory_reuse) {
+        reusable_across_network = false;
```

Comment:
Wouldn't it be better to force `ov::intel_gpu::enable_memory_pool` property in `void ExecutionConfig::apply_debug_options`?

Reply:
At first I thought it too, but maybe it will hide pool's allocation behavior itself, so I thought "wehther to reuse or not" and "whether to use pool or not" should be handled separately. how do you think?

Reply:
@yeonbok hmm, yeah, it makes sense, thanks for explanation!

Reply:
Thanks!